js/plugins/googleai/src/embedder.ts

```diff
@@ -69,6 +77,8 @@ export const SUPPORTED_MODELS = {
   'text-embedding-004': textEmbedding004,
 };
 
+const customDimensionsSupportedModelNames = [textEmbedding004.name];
```

Comment:
I don't think this is necessary. I don't think we want to maintain this mapping. Let the API return errors if the option is not supported.

Reply:
Indeed, based on your comment, I also feel that an API error would be the more natural approach.
I thought it was important to ensure that the issue is not hidden when the option is applied to an embedding model that does not support it.
Thank you for your comment.